### PR TITLE
Make improvements to ORA's development environment

### PR DESCRIPTION
**Pull request by andy-armstrong**
@dianakhuang and I have made a number of improvements to support an ORA development environment:
 - added support for using ORA in devstack
 - made it possible to debug ORA and its tests in PyCharm
 - made the XBlock use the raw JavaScript source files when in debug mode (versus the bundled JS)
 - added a .jshintrc file (I haven't added the code to use it yet)
 - documented the process: https://openedx.atlassian.net/wiki/display/TNL/ORA+FAQ

Please review: @dianakhuang @cahrens 

FYI: @efischer19 @symbolist @muhammad-ammar @muzaffaryousaf

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "edx-ora2",
-  "version": "0.0.1",
+  "version": "0.2.0",
```
Comment (cahrens) [inline package.json]: Why this version change?

File: setup.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def load_requirements(*requirements_paths):
 
 setup(
     name='ora2',
-    version='0.0.1',
+    version='0.2.0',
```
Comment (cahrens) [inline setup.py]: Why this version change?

File: settings/test_with_coverage.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"""
+Django settings for running tests with coverage.
```
Comment (cahrens) [inline settings/test_with_coverage.py]: Why do we want to separate test and test_with_coverage?

File: package.json (hunk 1 above)
Comment (andy-armstrong) [inline package.json]: I described this in the PR comment. It seemed broken that this had never been bumped. However, this came from the feature branch, so I could move it back as that's where the new functionality is coming.

File: settings/test_with_coverage.py (hunk 3 above)
Comment (andy-armstrong) [inline settings/test_with_coverage.py]: The issue is that all tests were run with coverage, even when running a single test in the debugger. This change makes the behavior more like edx-platform, where we only run coverage when doing everything at once, i.e. on Travis.

File: setup.py (hunk 2 above)
Comment (andy-armstrong) [inline setup.py]: To match the version in the package. I can revert this one too.

File: package.json (hunk 1 above)
Comment (cahrens) [inline package.json]: I think it is odd to bump it here, since there is no change in user functionality.

**Comment by cahrens** [comment]
:+1: after you back out the version number change

**Comment by andy-armstrong** [comment]
@cahrens I've removed the version number changes, squashed and pushed up again.

File: scripts/test.sh [hunk 4]
```diff
@@ -4,6 +4,7 @@
 set -e
 
 cd `dirname $BASH_SOURCE` && cd ..
+export DJANGO_SETTINGS_MODULE=${DJANGO_SETTINGS_MODULE:-settings.test_with_coverage}
```
Comment (dianakhuang) [inline scripts/test.sh]: Should `settings.test_with_coverage` be in quotes? `settings.test` is.

**Comment by dianakhuang** [comment]
:+1: otherwise

**Comment by symbolist** [comment]
@andy-armstrong Can the README be pointed to the instructions on the wiki? I don't think the orastack or workbench are of much use for doing any work on this repo.

**Comment by andy-armstrong** [comment]
@symbolist That's a good point. I'll clean up the documentation.

**Pull request merged**